pkg/migrator/migrator.go

```diff
@@ -0,0 +1,231 @@
+	associatedCscName := subscription.GetLabels()[builders.OwnerNameLabel]
+	possibleCsName := extractCsName(associatedCscName)
+	possibleCsNamespace := subscription.GetLabels()[builders.OwnerNamespaceLabel]
+	// try and fetch the CatalogSource
+	datastoreCs := &olm.CatalogSource{}
+	namespacedName := types.NamespacedName{Name: possibleCsName, Namespace: possibleCsNamespace}
+	err := client.Get(context.TODO(), namespacedName, datastoreCs)
+	if err != nil {
+		return nil, err
+	}
+	return datastoreCs, nil
+}
+
+// findCsFromOpsrc extracts the packageName from a Subscription,
+// and finds the corresponding CatalogSource that the package belongs to.
+func findCsFromOpsrc(subscription *olm.Subscription, kubeClient client.Client) (*olm.CatalogSource, error) {
+	packageName := subscription.Spec.Package
+	opsrcs := &v1.OperatorSourceList{}
+	err := kubeClient.List(context.TODO(), &client.ListOptions{}, opsrcs)
+	if err != nil {
+		return nil, err
+	}
+	for _, instance := range opsrcs.Items {
+		if !isPackageInOpsrc(packageName, &instance) {
+			continue
+		}
+		// fetch the CatalogSource with the same name
+		datastoreCs := &olm.CatalogSource{}
+		namespacedName := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
+		err = kubeClient.Get(context.TODO(), namespacedName, datastoreCs)
+		if err != nil {
+			return nil, err
+		}
+		return datastoreCs, nil
+	}
+	return nil, nil
+}
+
+// isPackageInOpsrc takes the name of a package, and an OperatorSource
+// as input, and returns true if the package is present in the OperatorSource's
+// status.Packages field
+func isPackageInOpsrc(packageName string, opsrc *v1.OperatorSource) bool {
+	packages := getOpsrcPackages(opsrc)
+	for _, pkg := range packages {
+		if pkg == packageName {
+			return true
+		}
+	}
+	return false
+}
+
+// extractCsName takes the name of a CatalogSourceConfig
+// as input, and infers the name of the corresponding
+// CatalogSource datastore. Installed CSCs follow the
+// following naming pattern: installed-publisher-namespace.
+// For example, for a CatalogSourceConfig named
+// `installed-community-openshift-operators`, extractCsName
+// extracts `community` off the name, appends `operators`
+// to it, and returns `community-operators` as output
+func extractCsName(cscName string) string {
+	possibleCsName := strings.Split(cscName, "-")[1]
+	return fmt.Sprintf("%s-%s", possibleCsName, "operators")
+}
+
+// getOpsrcPackages returns the list of packages from the Status block
+func getOpsrcPackages(opsrc *v1.OperatorSource) []string {
+	return strings.Split(opsrc.Status.Packages, ",")
+}
+
+// setSubsCatalogSource sets the Spec.source and Spec.sourceNamespace fields of a
+// Subscription, and returns the subscription
+func setSubsCatalogSource(subscription *olm.Subscription, csName string, csNamespace string) *olm.Subscription {
+	subscription.Spec.CatalogSource = csName
+	subscription.Spec.CatalogSourceNamespace = csNamespace
+	return subscription
```

Comment:
@aravindhp We want the subscription to be updated to the new values. 
Does that answer your question or am I misunderstanding the question?

Reply:
What I am trying to say is `updatedSub := setSubsCatalogSource(&instance, datastoreCs.GetName(), datastoreCs.GetNamespace())`  is equivalent to `setSubsCatalogSource(&instance, datastoreCs.GetName(), datastoreCs.GetNamespace())`. You can use `instance` instead of `updatedSub` given you are using a pointer. 

You might also want to consider if this function is really required given you are only calling it in one place and the amount of work that is being done in it.

Reply:
Ah, true. It made sense when I had accidentally modified the vendor code, but now that I'm not doing that the function doesn't make that much sense. I'll update the object at the call site.